### PR TITLE
Validate couper configuration

### LINKED ISSUE #188: Check for unknown attributes on config load
At the moment any typo or unknown attribute would be loaded without any hint for the user.

```hcl
request {
  path = "/"
  backend "ref" {}
```

`path` for example is not specified within this block and may lead to wrong assumptions.

correct:
```hcl
request {
  backend "ref" {
    path = "/"
}
```

related: #168

### PR DESCRIPTION
**Pull request by alex-schneider**
Closes #188 #168 

---
<details>
    <summary>Reviewer checklist</summary>
    <ul>
        <li>Read PR description: a summary about the changes is required</li>
        <li>Changelog updated</li>
        <li>Documentation: docs/{Reference, Cli, ...}, Docker and cli help/usage</li>
        <li>Pulled branch, manually tested</li>
        <li>Verified requirements are met</li>
        <li>Reviewed the code</li>
        <li>Reviewed the related tests</li>
    </ul>
</details>

### CODE REVIEW

File: command/verify.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	if len(args) != 1 {
+		v.Usage()
+		logger.Error(fmt.Errorf("invalid number of arguments given"))
+		return fmt.Errorf("")
+	}
+
+	_, err := configload.LoadFile(args[0], true)
+	if diags, ok := err.(hcl.Diagnostics); ok {
+		for _, err := range diags {
+			logger.Error(err)
```
Comment (malud) [inline command/verify.go:30]:
```suggestion
			logger.WithError(err).Error()
```
to apply log hooks

File: command/verify.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+func (v Verify) Execute(args Args, _ *config.Couper, logger *logrus.Entry) error {
+	if len(args) != 1 {
+		v.Usage()
+		logger.Error(fmt.Errorf("invalid number of arguments given"))
+		return fmt.Errorf("")
+	}
+
+	_, err := configload.LoadFile(args[0], true)
+	if diags, ok := err.(hcl.Diagnostics); ok {
+		for _, err := range diags {
```
Comment (malud) [inline command/verify.go:29]:
```suggestion
		for _, d := range diags {
```
shadowed `err`

File: config/error_handler.go [hunk 3]
```diff
@@ -11,7 +11,7 @@ var _ Inline = &ErrorHandler{}
 
 // ErrorHandler represents a subset of Endpoint.
 type ErrorHandler struct {
-	Kinds     []string
+	Kinds     []string  // FIXME: Why not `hcl:"type,label"`?
```
Comment (malud) [inline config/error_handler.go:14]: Just ask, labels have a fixed order and MUST be provided if defined within schema. Count MUST match and cant be one and then two etc. This implementation is flexible.
```suggestion
	Kinds     []string 
```

File: main.go [hunk 4]
```diff
@@ -88,7 +88,17 @@ func realmain(arguments []string) int {
 	}
 	env.Decode(&flags)
 
-	confFile, err := configload.LoadFile(flags.FilePath)
+	if cmd == "verify" {
+		log := newLogger(flags.LogFormat, flags.LogLevel, flags.LogPretty)
+
+		err := command.NewCommand(ctx, cmd).Execute(command.Args{flags.FilePath}, nil, log)
```
Comment (malud) [inline main.go:94]:
```suggestion
		err = command.NewCommand(ctx, cmd).Execute(command.Args{flags.FilePath}, nil, log)
```

Comment (malud) [reply main.go:94]: log returned err here with `WithError()` instead of returning empty ones.

**Review by malud — CHANGES_REQUESTED** [review]

File: main.go (hunk 4 above)
Comment (alex-schneider) [reply main.go:94]: No, the current IF block returns 1 or 0. The logger is never reached.

File: main_test.go [hunk 5]
```diff
@@ -42,6 +44,7 @@ func Test_realmain(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(subT *testing.T) {
+			fmt.Printf("%s\n", tt.name)
```
Comment (malud) [inline main_test.go:47]: rm

File: main_test.go [hunk 6]
```diff
@@ -53,6 +56,9 @@ func Test_realmain(t *testing.T) {
 			}
 			env.OsEnviron = os.Environ
 
+			// e := localHook.LastEntry()
```
Comment (malud) [inline main_test.go:59]: rm

**Review by malud — CHANGES_REQUESTED** [review]
lgtm so far.

**Review by malud — APPROVED** [review]

**Pull request merged**